inbm-lib/inbm_common_lib/utility.py

```diff
@@ -205,7 +205,9 @@ def is_within_directory(directory: str, target: str) -> bool:
     return prefix == abs_directory
 
 
-def safe_extract(tarball: tarfile.TarFile, path: str = ".", members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
+def safe_extract(tarball: tarfile.TarFile,
+                 path: str = ".",
+                 members: Optional[Iterable[tarfile.TarInfo]] = None, *, numeric_owner: bool = False) -> None:
     """Avoid path traversal when extracting tarball
 
     @param tarball: tarball to extract
```

Comment:
**Suggestion:** The safe_extract function should handle the case where members is provided but empty, which currently would lead to extracting the entire tarball, potentially causing unexpected behavior. [possible issue]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    if members is not None:
        members = list(members)  # Ensure it's not an empty iterator
        if not members:
            raise ValueError("The 'members' parameter is empty. No files to extract.")
    for member in (members if members is not None else tarball.getmembers()):
        member_path = os.path.join(path, member.name)
        if not is_within_directory(path, member_path):
            raise IOError("Attempted Path Traversal in Tar File")
    tarball.extractall(path, members, numeric_owner=numeric_owner)
```